java/java-impl/src/com/intellij/codeInsight/FunctionalInterfaceSuggester.java

```diff
@@ -251,11 +251,11 @@ else if (expression instanceof PsiMethodReferenceExpression) {
                   right[i + offset] = targetMethodParameters[i].getType();
                 }
 
-                left[parameters.length] = method.isConstructor() ? qualifierType : method.getReturnType();
+                left[parameters.length] = method.isConstructor() ? qualifierType : result.getSubstitutor().substitute(method.getReturnType());
```

Comment:
Is this correct? I noticed that if I change the code from [`InferenceSession`](https://github.com/JetBrains/intellij-community/blob/master/java/java-psi-impl/src/com/intellij/psi/impl/source/resolve/graphInference/InferenceSession.java#L90)
```java
      final PsiType rightType = mySiteSubstitutor.substitute(rightTypes[i]);
      PsiType t = substituteWithInferenceVariables(leftTypes[i]);
      PsiType s = substituteWithInferenceVariables(rightType);
```
to
```java
      final PsiType rightType = mySiteSubstitutor.substitute(rightTypes[i]);
      final PsiType leftType = mySiteSubstitutor.substitute(leftTypes[i]);
      PsiType t = substituteWithInferenceVariables(leftType);
      PsiType s = substituteWithInferenceVariables(rightType);
```
I can revert this change here, will it be correct?